### ISSUE TITLE
Error when trying to increase number of layers

### ISSUE THREAD

**Issue by chanedwin** [superMDguy/Variational-Recurrent-Autoencoder-Tensorflow#3]
Hey! Thanks for organising this repo, really appreciate it. 

I'm hitting a MatMul error when i increase num_layers in config.json to something more than 1 - ie more layers 

Error : 
tensorflow.python.framework.errors_impl.InvalidArgumentError: Dimensions must be equal, but are 300 and 1500 for 'variational_encoder_with_buckets/encoder_to_latent/MatMul' (op: 'MatMul') with input shapes: [?,300], [1500,32].

Entire Stack Trace :   
Traceback (most recent call last):
  File "vrae.py", line 539, in <module>
    tf.app.run()
  File "/home/edwin/anaconda3/lib/python3.6/site-packages/tensorflow/python/platform/app.py", line 48, in run
    _sys.exit(main(_sys.argv[:1] + flags_passthrough))
  File "vrae.py", line 536, in main
    train(config)
  File "vrae.py", line 176, in train
    model = create_model(sess, config, False)
  File "vrae.py", line 132, in create_model
    dtype=dtype)
  File "/home/edwin/projects/bot/Variational-Recurrent-Autoencoder-Tensorflow/seq2seq_model.py", line 240, in __init__
    softmax_loss_function=softmax_loss_function)
  File "/home/edwin/projects/bot/Variational-Recurrent-Autoencoder-Tensorflow/seq2seq.py", line 837, in variational_encoder_with_buckets
    mean, logvar = enc_latent(encoder_last_state)
  File "/home/edwin/projects/bot/Variational-Recurrent-Autoencoder-Tensorflow/seq2seq_model.py", line 183, in enc_latent_f
    dtype=dtype)
  File "/home/edwin/projects/bot/Variational-Recurrent-Autoencoder-Tensorflow/seq2seq.py", line 707, in encoder_to_latent
    mean_logvar = prelu(tf.matmul(encoder_state, w) + b)
  File "/home/edwin/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/math_ops.py", line 1891, in matmul
    a, b, transpose_a=transpose_a, transpose_b=transpose_b, name=name)
  File "/home/edwin/anaconda3/lib/python3.6/site-packages/tensorflow/python/ops/gen_math_ops.py", line 2437, in _mat_mul
    name=name)
  File "/home/edwin/anaconda3/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/home/edwin/anaconda3/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 2958, in create_op
    set_shapes_for_outputs(ret)
  File "/home/edwin/anaconda3/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 2209, in set_shapes_for_outputs
    shapes = shape_func(op)
  File "/home/edwin/anaconda3/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 2159, in call_with_requiring
    return call_cpp_shape_fn(op, require_shape_fn=True)
  File "/home/edwin/anaconda3/lib/python3.6/site-packages/tensorflow/python/framework/common_shapes.py", line 627, in call_cpp_shape_fn
    require_shape_fn)
  File "/home/edwin/anaconda3/lib/python3.6/site-packages/tensorflow/python/framework/common_shapes.py", line 691, in _call_cpp_shape_fn_impl
    raise ValueError(err.message)

**Comment by superMDguy** [comment]
I got the same error you did. I've been looking through the code and, strangely, it almost looks like multiple layers weren't supported in the original repository. Specifically, the encoder function [here](https://github.com/Chung-I/Variational-Recurrent-Autoencoder-Tensorflow/blob/master/seq2seq.py#L384) doesn't seem to take into account the `n_layers` parameter. I'll look into it more to see if I'm missing something, or try to implement it myself.

**Comment by chanedwin** [comment]
hey! yeah i think i fixed it but my fix is rather hacker-ish, not sure if it works for all edge cases and if it is even training properly

inside seq2seq_models.py ,
I used : 

```
    # Create the internal multi-layer cell for our RNN.
    cells = tf.contrib.rnn.MultiRNNCell([tf.contrib.rnn.GRUCell(size) for _ in range(num_layers)])
    if use_lstm:
      cells = tf.contrib.rnn.MultiRNNCell([tf.contrib.rnn.BasicLSTMCell(size) for _ in range(num_layers)])
    cell = cells
```

instead of 
```
    # Create the internal multi-layer cell for our RNN.
    single_cell = tf.nn.rnn_cell.GRUCell(size)
    if use_lstm:
      single_cell = tf.nn.rnn_cell.BasicLSTMCell(size)

    cell = single_cell
```

I'll test it more and push a commit if there are no bugs!

**Comment by superMDguy** [comment]
That looks great! When you're ready, feel free to push and create a pr, and I'll check it out and merge it.

**Comment by superMDguy** [comment]
The only potential issues I see are:
1. We'll should probably have something similar in the decoder function. It will work regardless, but it'd be nice to have multilayer support. You can add this to your pull request if you want, or I can do it.
2. I'm not sure if it will work with 1 layer, but I can try it out later.

Thanks for your help!

**Comment by chanedwin** [comment]
pull request at #4

**Comment by superMDguy** [comment]
Closed by #4
